src/main/java/gift/service/ProductService.java

```diff
@@ -46,30 +51,43 @@ public ProductListResponseDTO getAllProducts(int page, int size, String criteria
 
     @Transactional(readOnly = true)
     public ProductResponseDTO getOneProduct(Long productId) {
-        Product product = jpaProductRepository.findById(productId)
-            .orElseThrow(() -> new NoSuchElementException("id가 잘못되었습니다."));
+        Product product = getProduct(productId);
         return ProductResponseDTO.of(product);
     }
 
     public Long addProduct(ProductRequestDTO productRequestDTO) {
-        Product product = new Product(productRequestDTO.name(),
-            productRequestDTO.price(), productRequestDTO.imageUrl());
+        Category category = getCategory(productRequestDTO);
+
+        Product product = new Product(productRequestDTO.name(), productRequestDTO.price(),
+            productRequestDTO.imageUrl(), category);
+
         return jpaProductRepository.save(product).getId();
     }
 
     public Long updateProduct(Long productId, ProductRequestDTO productRequestDTO) {
-        Product product = jpaProductRepository.findById(productId)
-            .orElseThrow(() -> new NoSuchElementException("id가 잘못되었습니다."));
+        Product product = getProduct(productId);
+        Category category = getCategory(productRequestDTO);
 
         product.update(productRequestDTO.name(), productRequestDTO.price(),
-            productRequestDTO.imageUrl());
+            productRequestDTO.imageUrl(), category);
         return product.getId();
     }
 
     public Long deleteProduct(Long productId) {
-        Product product = jpaProductRepository.findById(productId)
-            .orElseThrow(() -> new NoSuchElementException("id가 잘못되었습니다."));
+        Product product = getProduct(productId);
         jpaProductRepository.delete(product);
         return product.getId();
     }
+
+    private Product getProduct(Long productId) {
+        Product product = jpaProductRepository.findById(productId)
+            .orElseThrow(() -> new NoSuchElementException("id가 잘못되었습니다."));
+        return product;
+    }
+
+    private Category getCategory(ProductRequestDTO productRequestDTO) {
+        Category category = jpaCategoryRepository.findById(productRequestDTO.categoryId())
+            .orElseThrow(() -> new NoSuchElementException("id가 잘못되었습니다."));
+        return category;
+    }
```

Comment:
요런 친구들은 그냥 빠르게 리턴해도 좋을 것 같네요!

```suggestion
private Product getProduct(Long productId) {
        return jpaProductRepository.findById(productId)
            .orElseThrow(() -> new NoSuchElementException("id가 잘못되었습니다."));
    }

    private Category getCategory(ProductRequestDTO productRequestDTO) {
        return jpaCategoryRepository.findById(productRequestDTO.categoryId())
            .orElseThrow(() -> new NoSuchElementException("id가 잘못되었습니다."));
    }
```

Reply:
넵!! 감사합니다